naucse/routes.py

```diff
@@ -182,6 +184,20 @@ def subpage_url(page_slug):
     kwargs.setdefault('title', page.title)
     kwargs.setdefault('content', content)
 
+    if (page.subpages != None and page.slug in page.subpages.keys()
+    and "prev" in page.subpages[page.slug].keys() and "next" in page.subpages[page.slug].keys()) :
+        prev_segments = page.subpages[page.slug]["prev"].split(':')
+        next_segments = page.subpages[page.slug]["next"].split(':')
+
+        if len(next_segments) == 2:
+            next_segments.append("index")
+
+        if len(prev_segments) == 2:
+            prev_segments.append("index")
+
+        kwargs['prv'] = models.Navigation(prev_segments[0], prev_segments[1], prev_segments[2])
+        kwargs['nxt'] = models.Navigation(next_segments[0], next_segments[1], next_segments[2])
```

Comment:
All the code to create a `Navigation` is duplicated twice here. Is there a better place for initialization code to go?